### PR TITLE
[arrow] Update to 18.1.0

### LINKED ISSUE #42369: arrow update to 18.1.0
### Library name

arrow

### New version number

18.1.0

### Other information that may be useful (release notes, etc...)

https://github.com/apache/arrow/releases/tag/apache-arrow-18.1.0

### PR DESCRIPTION
**Pull request by amoeba**
Fixes https://github.com/microsoft/vcpkg/issues/42369
- [x] Changes comply with the [maintainer guide](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/contributing/maintainer-guide.md).
- [x] SHA512s are updated for each updated download.
- [x] The "supports" clause reflects platforms that may be fixed by this new version.
- [x] ~~Any fixed [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt) entries are removed from that file.~~
- [x] ~~Any patches that are no longer applied are deleted from the port's directory.~~
- [x] The version database is fixed by rerunning `./vcpkg x-add-version --all` and committing the result.
- [x] Only one version is added to each modified port's versions file.

### CODE REVIEW

**Comment by amoeba** [comment]
@microsoft-github-policy-service agree

**Comment by LilyWangLL** [comment]
Please fix the build error of `arrow:x64-android` and `arrow:arm64-android`, you could download the faiure log from https://dev.azure.com/vcpkg/public/_build/results?buildId=109678&view=artifacts&pathAsName=false&type=publishedArtifacts.

**Comment by amoeba** [comment]
I've filed https://github.com/apache/arrow/issues/44886 to track the process of figuring out a fix. I'll post an update here once we have a fix.

**Comment by dg0yt** [comment]
Note that vcpkg CI builds Android with API level 21. At that level, C runtime is very incomplete. It might be necessary to record an expected fail in scripts/ci.baseline.txt.

**Comment by amoeba** [comment]
I pushed up a patch for the Android failure and will keep an eye on CI now.

File: ports/arrow/portfile.cmake [hunk 1]
```diff
@@ -2,16 +2,17 @@ vcpkg_download_distfile(
...
     PATCHES
-        android.patch
-        msvc-static-name.patch
-        utf8proc.patch
-        thrift.patch
+        0001_msvc-static-name.patch
+        0002_thrift.patch
+        0003_utf8proc.patch
+        0004_android.patch
+        0005_android.patch
```
Comment (dg0yt) [inline ports/arrow/portfile.cmake:15]: Patch names :face_holding_back_tears:

Comment (amoeba) [reply ports/arrow/portfile.cmake:15]: This is good yeah?

Comment (amoeba) [reply ports/arrow/portfile.cmake:15]: Just curious if you're requesting a change here or not.

Comment (dg0yt) [reply ports/arrow/portfile.cmake:15]: I'm not in the position to *request* changes, but IMHO it is not good.

Comment (amoeba) [reply ports/arrow/portfile.cmake:15]: Are you able to give any guidance on what would be good?

Comment (dg0yt) [reply ports/arrow/portfile.cmake:15]: e.g. `0006-android-datetime.patch`.

(But I still wonder if patching a vendored datetime is the right thing to start with.)

Comment (amoeba) [reply ports/arrow/portfile.cmake]: Re: the filenames, I pushed changes to the filenames.

**Comment by amoeba** [comment]
@LilyWangLL we added a patch to the portfile and that looks like it's fixed the two failing jobs from before. I'm seeing the `microsoft.vcpkg.pr (x86 windows)` job fail but it looks related. Can you take a look and let us know if there's anything else that needs to be done here?

By the way, we plan to upstream the patch and remove it from this port in a future update, see https://github.com/apache/arrow/issues/43390.

File: versions/a-/arrow.json [hunk 2]
```diff
@@ -1,5 +1,10 @@
 {
   "versions": [
+    {
+      "git-tree": "8b0c4c21d05a62cc0b626e93149882cd127f83f8",
```
Comment (dg0yt) [inline versions/a-/arrow.json:4]:
```suggestion
      "git-tree": "cff712c97f67ce2ef9061d823c751f5e209e5838",
```
The x86-windows port build suceeds, but that triplet also checks these entries which are normally managed with `vcpkg x-add-version [--overwrite-version] arrow`.

Comment (amoeba) [reply versions/a-/arrow.json:4]: Ah, thanks. I now understand that better (should've been running that as I amended the PR).

**Comment by BillyONeal** [comment]
Thanks for the update!

**Comment by amoeba** [comment]
Thanks for the merge @BillyONeal and for the reviews @dg0yt.

**Pull request merged**